src/realm/object-store/schema.cpp

```diff
@@ -58,9 +58,10 @@ Schema::Schema(base types) noexcept
 
 Schema::iterator Schema::find(StringData name) noexcept
 {
-    auto it = std::lower_bound(begin(), end(), name, [](ObjectSchema const& lft, StringData rgt) {
-        return lft.name < rgt;
-    });
+    auto it = std::lower_bound(begin(), end(), std::string_view(name.data(), name.size()),
```

Comment:
instead of constructing a string_view from the data()/size() results, we could also do static_cast<std::string_view>(name) and call the conversion operator - except that looking at that operator, it's broken because it assumes that m_data is a C-string 😭 

Can we go ahead and fix that here as well?